main.tf

```diff
@@ -660,6 +660,30 @@ resource "aws_s3_bucket_replication_configuration" "default" {
             replica_kms_key_id = rule.value.replica_kms_key_arn
           }
         }
+
+        dynamic "metrics" {
```

Comment:
please ensure that dynamic blocks are used consistently. E.g. take a look at source_selection_criteria


dynamic "metrics" {
 for_each = rule.value.metrics != null ? { create = true } : {}

          content {
            status = rule.value.metrics.status ? "Enabled" : "Disabled"

            event_threshold {
              minutes = rule.value.metrics.event_threshold_minutes
            }
          }
        }